### PR TITLE
adding `db.schema.createTable(name).modifyFront(modifier)` & `db.schema.createTable(name).modifyEnd(modifier)`.

### PR DESCRIPTION
**Pull request by igalklebanov**
`CREATE TABLE` statement exists in all 3 base dialects ([postgres](https://www.postgresql.org/docs/current/sql-createtable.html), [mysql](https://dev.mysql.com/doc/refman/8.0/en/create-table.html), [sqlite](https://www.sqlite.org/lang_createtable.html)) and their variants (cockroachdb, planetscale, singlestore). Kysely supports compiling such queries using `SchemaModule` (`db.schema.createTable(name)...`).

Dialects and their variants provide abilities to choose table type/scope and configure your table beyond the defaults (e.g. engine, collation, compression, partitioning, etc.). You do this between `CREATE` & `TABLE` keywords and after column, constraint and index scope. Kysely supports some common use cases like temporary tables (`db.schema.createTable(name).temporary()`), and on commit. There's a lot of dialect-specific stuff here and it probably won't be beneficial to implement most of it in Kysely as functions. Some will probably end up as functions in Kysely, but it'll take time.

Consumers who need more configurability when creating tables are blocked from using `SchemaModule`. E.g. in singlestore (a "MySQL variant", formerly memSQL), table storage types are at the core of the product. You choose between a columnstore table and an in-memory rowstore table right after the `CREATE` keyword. E.g. you'd like to partition your table by region.

----

tldr

This PR adds the ability to: 
- inject custom SQL right after the `CREATE` keyword, and before `TABLE` keyword using:

```typescript
import { sql } from 'kysely'

db.schema
    .createTable('my_table')
    .modifyFront(sql`global temporary`)
    .addColumn('id', 'integer', col => col.primaryKey())
```

```sql
create global temporary table "my_table" (
    "id" integer primary key
)
```

- inject custom SQL right at the end of `CREATE ... TABLE ( ... ) ...` statements using:

```typescript
import { sql } from 'kysely'

db.schema
    .createTable('my_table')
    .addColumn('id', 'integer', col => col.primaryKey())
    .modifyEnd(sql`collate utf8_unicode_ci`)
```

```sql
create table `my_table` (
  `id` integer primary key
) collate utf8_unicode_ci
```

### CODE REVIEW

**Comment by igalklebanov** [comment]
Why is it the query builder's job, and not... the API's job to json schema validate rhe payload?!

File: src/operation-node/operation-node-transformer.ts [hunk 1]
```diff
@@ -390,6 +390,8 @@ export class OperationNodeTransformer {
       temporary: node.temporary,
       ifNotExists: node.ifNotExists,
       onCommit: node.onCommit,
+      frontModifiers: node.frontModifiers,
```
Comment (koskimas) [inline src/operation-node/operation-node-transformer.ts:393]: You need to transform these `this.transformNodeList(node.frontModifiers)`

File: src/operation-node/operation-node-transformer.ts [hunk 2]
```diff
@@ -390,6 +390,8 @@ export class OperationNodeTransformer {
...
+      endModifiers: node.endModifiers,
```
Comment (koskimas) [inline src/operation-node/operation-node-transformer.ts:394]: Same here

Comment (igalklebanov) [reply src/operation-node/operation-node-transformer.ts]: Done!

File: src/operation-node/operation-node-transformer.ts (hunk 1 above)
Comment (igalklebanov) [reply src/operation-node/operation-node-transformer.ts]: Done!

**Review by koskimas — APPROVED** [review]
LGTM! Just needs a rebase

**Comment by koskimas** [comment]
Still needs a rebate 😀

**Comment by igalklebanov** [comment]
Made a mess, sorry.. 💩 

Hope this is good to go now.. 🤞
